### DISCUSSION TITLE
Is it possible to get the timestamps of each word?

### DISCUSSION THREAD

**Discussion by Xpl0itU** [coqui-ai/TTS#2614] [General Q&A]
Let’s say that I want to generate a TTS audio from some text, and I’d like to get the timestamp for each word, is it possible to do it, if so, how?

**Comment by atommerlin** [comment]
It's definitely a workaround, but you could use Whisper, as it outputs timestamps.

**Comment by p0p4k** [comment]
Yes, we can get timestamps. You can refer to this PR by me https://github.com/coqui-ai/TTS/pull/2526

**Reply by p0p4k** [reply]
I even go in detail to change the time stamp for custom durations and create new unseen speakers from a trained model.

**Reply by knochenhans** [reply]
Hi, this looks very promising, but I’m a little confused about the state of the PR seeing it is marked closed. Will the code changes be included in one the coming coqui-ai releases or will this remain a custom change? Sorry if this is a dumb question, I’m not all that familiar with git workflows 😅

**Reply by p0p4k** [reply]
@knochenhans no that code will not be merged for the inconvenience of making tests and handling notebooks in future releases. However, you can make a new branch on your local copy and just do those changes. While you are on GitHub, it might be a good idea to learn a few things here so you don't get confused in navigating through so many wonderful repositories of code!
